File: Python/rob.py
def rob(nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        if len(nums)<=2:
            return max(nums)
        a,b = nums[0],max(nums[0],nums[1])
        for i in xrange(2,len(nums)):
            tmp = b
            b = max(nums[i]+a, b)
            a = tmp
        return b
        

File: src/S0198_House Robber/rob.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function (nums) {
  if (!nums.length) return 0;
  if (nums.length <= 2) return Math.max.apply(null, nums);

  let start = nums[0];
  let max = Math.max(nums[0], nums[1]);

  for (let i = 2; i < nums.length; i++) {
    const tmp = max;
    max = Math.max(nums[i] + start, max);
    start = tmp;
  }
  return max;
};
